src/Shared/constants.ts

```diff
@@ -24,6 +24,14 @@ export const SERVER_SUBEXT = ".server";
 export const CLIENT_SUBEXT = ".client";
 export const MODULE_SUBEXT = "";
 
+export const FILENAME_WARNINGS = new Map();
+
+for (const scriptType of [SERVER_SUBEXT, CLIENT_SUBEXT, MODULE_SUBEXT]) {
+	for (const fileType of [TS_EXT, TSX_EXT, DTS_EXT]) {
+		FILENAME_WARNINGS.set(INIT_NAME + scriptType + fileType, INDEX_NAME + scriptType + fileType);
+	}
+}
+
```

Comment:
Iterating over array literals is a bit of a "code smell"... I wonder if there's a better way to do this.

Reply:
On one hand yeah, it's not great, but at the same time, I think it's fine here to create each possible combination of the specified string parts.